packages/report-server/src/__tests__/reportBuilder/transform/parser/functions.test.ts

```diff
@@ -0,0 +1,172 @@
+        expect(new TransformParser().evaluate("=dateStringToPeriod('2020-02-15', 'WEEK')")).toBe(
+          '2020W07',
+        ));
+    });
+
+    describe('formatAsFractionAndPercentage', () => {
+      it('formats values as fraction and percentage', () =>
+        expect(new TransformParser().evaluate('=formatAsFractionAndPercentage(1, 10)')).toBe(
+          '1/10 = 10%',
+        ));
+
+      it('throws error if denominator is 0', () =>
+        expect(() => {
+          new TransformParser().evaluate('=formatAsFractionAndPercentage(1, 0)');
+        }).toThrow());
+    });
+  });
+
+  describe('math', () => {
+    describe('add', () => {
+      it('returns first number if second is undefined', () =>
+        expect(new TransformParser().evaluate('=1 + undefined')).toBe(1));
+
+      it('returns second number if first is undefined', () =>
+        expect(new TransformParser().evaluate('=undefined + 5')).toBe(5));
+
+      it('returns undefined if all values are undefined', () =>
+        expect(new TransformParser().evaluate('=undefined + undefined')).toBe(undefined));
+    });
+
+    describe('divide', () => {
+      it('returns undefined if second number is undefined', () =>
+        expect(new TransformParser().evaluate('=1 / undefined')).toBe(undefined));
+
+      it('returns undefined if first number is undefined', () =>
+        expect(new TransformParser().evaluate('=undefined / 5')).toBe(undefined));
+
+      it('returns undefined if both numbers are undefined', () =>
+        expect(new TransformParser().evaluate('=undefined / undefined')).toBe(undefined));
+    });
+
+    describe('sum', () => {
+      it('throws error for non-numeric input', () =>
+        expect(() => new TransformParser().evaluate("=sum('cat')")).toThrow());
+
+      it('throws error for non-numeric array input', () =>
+        expect(() => new TransformParser().evaluate("=sum([1, 'cat'])")).toThrow());
+
+      it('throws error for non-numeric inputs', () =>
+        expect(() => new TransformParser().evaluate("=sum(1, 'cat')")).toThrow());
+
+      it('returns undefined if input is undefined', () =>
+        expect(new TransformParser().evaluate('=sum(undefined)')).toBe(undefined));
+
+      it('returns undefined if input is array of undefined', () =>
+        expect(new TransformParser().evaluate('=sum([undefined, undefined])')).toBe(undefined));
+
+      it('returns undefined if inputs are undefined', () =>
+        expect(new TransformParser().evaluate('=sum(undefined, undefined)')).toBe(undefined));
+
+      it('returns sum if input is at least partially defined array', () =>
+        expect(new TransformParser().evaluate('=sum([undefined, 1])')).toBe(1));
+
+      it('returns sum if inputs are at least partially defined', () =>
+        expect(new TransformParser().evaluate('=sum(undefined, 1)')).toBe(1));
+
+      it('returns input if input is defined', () =>
+        expect(new TransformParser().evaluate('=sum(1)')).toBe(1));
+    });
```

Comment:
It seems very trivial but I suggest adding a test case for summing 2 numbers, and one for summing multiple (eg 4-5) numbers

Reply:
Good call, will add!